### DISCUSSION TITLE
v6.0.0

### DISCUSSION THREAD

**Discussion by paradajz** [shanteacontrols/OpenDeck#104] [New releases]
## New features

* Added initial support for DMX transmitter on all boards. The firmware implements Enttec USB Pro Widget API which makes the boards compatible with Open Lighting Architecture (OLA) through `usbserial` plugin.
* Implemented dual USB interface: MIDI and CDC. Both interfaces are now loaded at the same time. CDC interface can be used either for DMX or touchscreen programming (the board effectively becomes serial to USB converter). Standalone CDC mode is removed as there is no more need for it.
* Added new SysEx errors: UART Allocation Error and CDC Allocation Error. Shown in Web UI when attempting to initialize peripheral sharing the same channel with another peripheral which is already initialized.
* USB unique ID is now shown for AVR-based boards
* Added support for one more UART channel on Arduino Mega board - touchscreen is now moved to channel 2 by default

## Improvements

* Simplified and reworded some YAML keys for target configuration

## Fixes

* Fixed compilation for custom boards which aren't using any analog components
* Fixed incorrect touchscreen icon handling on boards supporting both LEDs and touchscreen
* Fixed some LEDs being slightly lit while not actually turned on for some boards (eg. official OpenDeck board)
* Fixed turning LEDs off when configured to react on Program Change messages
* Fixed LED state not being correctly set when switching presets

## Misc

* Changed the touchscreen baud rate to 38400 to avoid issues on AVR boards
* Target configuration is now moved to config/target directory
* Added example schematic for adding DMX transmitter ability to the boards
* Temporary _sysex.bin files aren't attached to the release anymore (regression from v5.5.5)
* Reorganized board directory
* Refactored MCU support - MCU constants now have configuration YAML files
* Refactored system initialization to make it more testable

<hr /><em>This discussion was created from the release <a href='https://github.com/shanteacontrols/OpenDeck/releases/tag/v6.0.0'>v6.0.0</a>.</em>

**Comment by js-midi** [comment]
After loading the firmware version on the mega2560, communication with the board is not established in the Open Deck Configurator.

**Reply by paradajz** [reply]
you need to update both mcus

**Reply by js-midi** [reply]
of course i did both
But only the rx LED is blinking

**Reply by paradajz** [reply]
I don't know. It works fine on multiple boards I have here.

**Reply by paradajz** [reply]
You need to make sure your fuse settings are correct.

**Reply by js-midi** [reply]
I always used microchip Studio to upload firmware.
It worked fine in other versions, but it doesn't work only in version 6.0.0
I am not good at using Linux or Windows terminal.

**Reply by js-midi** [reply]
Can you tell me how to set the fuse?

**Reply by paradajz** [reply]
You need to use the scripts I provide - terminal stuff. Sorry, I don't support graphical tools. It's all explained in the wiki.

**Reply by js-midi** [reply]
I'll try

**Reply by js-midi** [reply]
Windows 10 with WSL 1 - WSL2 won't work! (Ubuntu 20.04+)*
Can you easily tell me what exactly you mean?

**Reply by js-midi** [reply]
Please tell me the easiest and most comfortable system to use to build and update firmware.
It's a bit difficult to set up on Windows 10, which I'm currently using.

**Reply by paradajz** [reply]
I'm sorry but I have limited time. It's all explained in the wiki.

**Reply by js-midi** [reply]
yes i know
Thank you for your kind help so far.

**Comment by js-midi** [comment]
[1] I have the same symptoms on the discovery board
     Unable to enter OpenDeck Configuratoator after loading 6.0.0
    I used the firmware update function of Open Deck Configuratoator


[2] Can't mega2560 enter bootload mode of opendeck configurator?


[3]It works fine in stm32f411 blackfill

**Reply by paradajz** [reply]
All the boards work fine. The problem is at your end.
